### DISCUSSION TITLE
How do you add a custom endpoint to the REST API of a Dockerized MLFlow Model?

### DISCUSSION THREAD

**Discussion by bgalvao** [mlflow/mlflow#6247] [Q&A] [answered]
Let's say I want to add an endpoint called `/ale_explain` (based on [this](https://docs.seldon.io/projects/alibi/en/stable/examples/ale_classification.html#Calculate-Accumulated-Local-Effects)). It only takes a `predict_proba` function from a scikit classifier.

How could I subclass/extend MLFlow's classes in order to have this endpoint automatically generated in an MLFlow Model Docker image?

**Comment by JHibbard** [comment, marked as answer]
Hi bgalvao,

I think you're looking for the [MLflow deployments](https://www.mlflow.org/docs/latest/python_api/mlflow.deployments.html) feature. The answer will vary depending on the serving tool you're deploying to, but many support specifying an `endpoint` which would become the equivalent of your `/ale_explain` endpoint without needing to modify MLflow. Some serving tools support defining an [explain](https://www.mlflow.org/docs/latest/python_api/mlflow.deployments.html#mlflow.deployments.BaseDeploymentClient.explain) endpoint too.

Hope that helps!
Jim

**Reply by bgalvao** [reply]
Any example you could point me to? The API is a little too abstract. It seems like I would first create a deployment, then obtain a client to that deployment and from there add endpoints manually with [`create_endpoint`](https://www.mlflow.org/docs/latest/python_api/mlflow.deployments.html#mlflow.deployments.BaseDeploymentClient.create_endpoint). But all you can pass to that function is a `config` dict, and I am not sure how this dict could point to the ALE explainer binary (or something else, like a drift detector) to generate explanations.
